aws/resource_aws_cloudwatch_event_rule.go

```diff
@@ -123,7 +130,12 @@ func resourceAwsCloudWatchEventRuleCreate(d *schema.ResourceData, meta interface
 	}
 
 	d.Set("arn", out.RuleArn)
-	d.SetId(*input.Name)
+	id := aws.StringValue(input.Name)
+	eventBusName := aws.StringValue(input.EventBusName)
+	if len(eventBusName) > 0 {
+		id = eventBusName + "-" + id
```

Comment:
I suggest using `/` as the separator as it's not a valid event bus name character while `-` is.
Also don't prefix the name if it's `default` (see https://github.com/terraform-providers/terraform-provider-aws/pull/10256/files#r382815323).